pkg/microservice/aslan/core/common/service/instantmessage/service.go

```diff
@@ -295,19 +295,23 @@ func (w *Service) createNotifyBodyOfWorkflowIM(weChatNotification *wechatNotific
 				if err != nil {
 					return "", "", nil, err
 				}
-				branch, commitID, commitMsg, gitCommitURL := "", "", "", ""
+				branchTag, branchTagType, commitID, commitMsg, gitCommitURL := "", "Branch", "", "", ""
 				for idx, buildRepo := range buildSt.JobCtx.Builds {
 					if idx == 0 || buildRepo.IsPrimary {
-						branch = buildRepo.Branch
+						branchTag = buildRepo.Branch
+						if buildRepo.Tag != "" {
+							branchTagType = "Tag"
+							branchTag = buildRepo.Tag
+						}
 						if len(buildRepo.CommitID) > 8 {
 							commitID = buildRepo.CommitID[0:8]
 						}
 						commitMsgs := strings.Split(buildRepo.CommitMessage, "\n")
 						if len(commitMsgs) > 0 {
 							commitMsg = commitMsgs[0]
 						}
-						if len(commitMsg) > 40 {
-							commitMsg = commitMsg[0:40]
+						if len(commitMsg) > 60 {
```

Comment:
These numbers are better to be characterized by constants or service parameters.